core/consensus/src/trace.rs

```diff
@@ -18,9 +18,21 @@ impl MetricTracer {
     }
 }
 
+fn err() -> ProtocolError {
+    ConsensusError::Other("tracing: failed when parse point".to_string()).into()
+}
+
+fn to_trace_str(point: TracePoint) -> ProtocolResult<String> {
+    let mut json = to_value(point).map_err(|_| err())?;
```

Comment:
It is a bad idea to swallow the original error information here.
There are 3 places you use `err()` function here. If error does happens, the `Err(e) => error!("tracing: convert json error {:?}", e),` only print `tracing: failed when parse point` for you, no other useful information to help debugging.
If you think error should not happen, use except, otherwise try to wrap the original error with necessary information.

Reply:
OK, I see, thx for the suggestion. I have make the error more detailed